pkg/provisioner/templates/containerd.go

```diff
@@ -32,29 +32,54 @@ const containerdTemplate = `
 with_retry 3 10s sudo apt-get update
 install_packages_with_retry ca-certificates curl gnupg -y
 sudo install -m 0755 -d /etc/apt/keyrings
-curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg
-sudo chmod a+r /etc/apt/keyrings/docker.gpg
 
-# Add the repository to Apt sources:
-echo \
-  "deb [arch="$(dpkg --print-architecture)" signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu \
-  "$(. /etc/os-release && echo "$VERSION_CODENAME")" stable" | \
-  sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
-  with_retry 3 10s sudo apt-get update
+# Download containerd 2.0 from GitHub
+ARCH=$(dpkg --print-architecture)
+echo "Detected architecture: $ARCH"
 
-# Install containerd
-install_packages_with_retry containerd.io=${CONTAINERD_VERSION}-1
+CONTAINERD_TAR="containerd-${CONTAINERD_VERSION}-linux-${ARCH}.tar.gz"
+CONTAINERD_URL="https://github.com/containerd/containerd/releases/download/v${CONTAINERD_VERSION}/${CONTAINERD_TAR}"
 
-# Configure containerd and start service
-mkdir -p /etc/containerd
+echo "Downloading containerd from: $CONTAINERD_URL"
+curl -fsSL -o ${CONTAINERD_TAR} ${CONTAINERD_URL}
+sudo tar Cxzvf /usr/local ${CONTAINERD_TAR}
+rm -f ${CONTAINERD_TAR}
+
+# Install runc
+RUNC_VERSION="1.2.5"
```

Comment:
This isn't really a dependency that I want to keep up to date. Is there a "latest"?

Reply:
Done